### DISCUSSION TITLE
where is the `edge_norm` argument of `RGCNConv` now ?

### DISCUSSION THREAD

**Discussion by RenzeLou** [pyg-team/pytorch_geometric#3688] [Q&A]
Dear authors,

I found an argument of the `RGCNConv` class in the old version of PyG (v 1.3.0), namely `edge_norm` (as shown below):
![image](https://user-images.githubusercontent.com/59914581/145711858-1aad159d-b109-4edb-8b21-4f9bb17ade4a.png)
I guess this argument should be the **problem-specific normalization constant** as claimed in equation (2) of the original paper:
![A( {QFO79CPT%PZLVWTOM L](https://user-images.githubusercontent.com/59914581/145712006-1c253243-12c6-4ea7-a006-9646aa742cba.png)

However, I didn't find it in the latest version of PyG (v 2.0.2). It seems like this API has changed a lot.

![image](https://user-images.githubusercontent.com/59914581/145712193-2f546ed5-3165-42f1-9ba8-b416f14d822e.png)

So, how can I set the `edge_norm` now (i.e., the latest version of PyG)?

**Comment by rusty1s** [comment]
Yes, the `edge_norm` attribute was removed in favor of supporting different `aggr` methods (`mean`, `add`, `max`), which should model most of the use-cases of `edge_norm`. The goal was to make applying mean aggregation much more easier :)

**Reply by RenzeLou** [reply]
Thank you for your kind reply！

Nevertheless, I want to use the `edge_norm` to make the weighted aggrgation (i.e., set different normalization constants dynamically). Does the current version of PyG still support this operation?

**Reply by rusty1s** [reply]
No it does not, I'm sorry. We need to add it back in. Let me know if you have interest in working with us together on this.

**Reply by RenzeLou** [reply]
Ok, maybe I can try.

**Comment by sailist** [comment]
Hi, I am now reimplement a paper which use `edge_norm` attribute. 

If given 
```python
out = self.conv1(x, edge_index, edge_type, edge_norm)
```

Is there any equal method I can workaround? Like
```python
out = self.conv1(x, edge_index, edge_type)
out = norm(out, edge_norm)
```

**Reply by rusty1s** [reply]
Currently no, we will likely bring this feature back as part of our `RGCN` refactor. Please keep reminding me :)

**Reply by rusty1s** [reply]
Relevant dicussion: https://github.com/pyg-team/pytorch_geometric/discussions/5217
